sale_payment_term_interest/tests/test_sale_order.py

```diff
@@ -0,0 +1,157 @@
+        interest_lines = set()
+        for line in sale.order_line:
+            if line.interest_line:
+                interest_lines.add(line)
+        self.assertEqual(len(interest_lines), 1)
+        line = interest_lines.pop()
+        self.assertAlmostEqual(line.price_subtotal, 3.82)
+        self.assertAlmostEqual(sale.amount_total, 253.82)
+        sale.check_interest_line()  # no error
+
+    def test_interest_change_line(self):
+        product1 = self.env.ref('product.product_product_7')
+        product2 = self.env.ref('product.product_product_9')
+        sale = self.env['sale.order'].create({
+            'partner_id': self.env.ref('base.res_partner_2').id,
+            'payment_term': self.payment_term.id,
+        })
+        self.env['sale.order.line'].create({
+            'product_id': product1.id,
+            'product_uom_qty': 1,
+            'product_uom': product1.uom_id.id,
+            'price_unit': 50,
+            'order_id': sale.id,
+        })
+        line2 = self.env['sale.order.line'].create({
+            'product_id': product2.id,
+            'product_uom_qty': 2,
+            'product_uom': product1.uom_id.id,
+            'price_unit': 100,
+            'order_id': sale.id,
+        })
+        self.assertEqual(len(sale.order_line), 2)
+        with self.assertRaises(exceptions.Warning):
+            sale.check_interest_line()
+
+        sale.update_interest_line()
+        sale.check_interest_line()  # no error
+        interest_line = sale._get_interest_line()
+        self.assertAlmostEqual(interest_line.price_subtotal, 3.82)
+        line2.product_uom_qty = 3
+
+        sale.update_interest_line()
+        sale.check_interest_line()  # no error
+        self.assertAlmostEqual(interest_line.price_subtotal, 5.35)
+
+        sale.payment_term = self.env.ref('account.account_payment_term_15days')
+        interest_line = sale._get_interest_line()
+        self.assertFalse(interest_line)
+
+    def test_interest_with_tax(self):
+        product1 = self.env.ref('product.product_product_7')
+        product2 = self.env.ref('product.product_product_9')
+        product_interest = self.env.ref('sale_payment_term_interest.'
+                                        'product_product_sale_order_interest')
+        tax = self.env['account.tax'].create({
+            'name': 'Percent tax',
+            'type': 'percent',
+            'amount': '0.1',
+        })
+        product_interest.taxes_id = [(6, 0, [tax.id])]
+        line1_values = {
+            'product_id': product1.id,
+            'product_uom_qty': 1,
+            'product_uom': product1.uom_id.id,
+            'price_unit': 50,
+            'tax_id': [(6, 0, [tax.id])],
+        }
+        line2_values = {
+            'product_id': product2.id,
+            'product_uom_qty': 2,
+            'product_uom': product1.uom_id.id,
+            'price_unit': 100,
+            'tax_id': [(6, 0, [tax.id])],
+        }
```

Comment:
The beggining of each test is almost the same here in the 3 cases. You could move what is repeated to setUp and then just write in your objects what changes from one unit test to the other.